### PR TITLE
layers:VkRayTracingPipelineCreateInfo* validation

### PR DESCRIPTION
**Pull request by souravpNV**

### CODE REVIEW

File: layers/parameter_validation_utils.cpp [hunk 1]
```diff
@@ -3903,6 +3974,79 @@ bool StatelessValidation::manual_PreCallValidateCreateRayTracingPipelinesKHR(VkD
...
+            skip |=
+                LogError(device, "VUID-VkRayTracingPipelineCreateInfoKHR-flags-02904",
+                         "vkCreateRayTracingPipelinesKHR(): flags must not include VK_PIPELINE_CREATE_INDIRECT_BINDABLE_BIT_NV.");
+        }
+        if (pCreateInfos[i].flags & VK_PIPELINE_CREATE_LIBRARY_BIT_KHR) {
+            if (pCreateInfos[i].pLibraryInterface == NULL)
+                skip |= LogError(device, "VUID-VkRayTracingPipelineCreateInfoKHR-flags-03465",
+                                 "If flags includes VK_PIPELINE_CREATE_LIBRARY_BIT_KHR, pLibraryInterface must not be NULL.");
+        }
+        for (uint32_t groupIndex = 0; groupIndex < pCreateInfos[i].groupCount; ++groupIndex) {
```
Comment (mark-lunarg) [inline layers/parameter_validation_utils.cpp:3987]: local vars should be lower_camel_case.

**Comment by mark-lunarg** [comment]
Running through CI now.

File: layers/parameter_validation_utils.cpp (hunk 1 above)
Comment (souravpNV) [reply layers/parameter_validation_utils.cpp:3987]: @MarkY-LunarG  isn't groupIndex already in camel case?  I see similar usage in auto generated paramter_validation.cpp

Comment (mark-lunarg) [reply layers/parameter_validation_utils.cpp:3987]: Sorry about that, I meant lower_snake_case.  Yeah, that's why we're trying to hard to makie these things more consistent.

**Comment by souravpNV** [comment]
@mark-lunarg :  just checking this now. In my last split, I reordered the history, that's why it shows there commits too.
Can you selectively  cherry pick my last change: 43f1c81  and merge to master

For the test https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/1725, similarly just pull the test content, some test are failing due to KHR->NV VUID change, which I had already fixed but are not present in paramater_validation_util.cpp change for that pull request.

**Comment by souravpNV** [comment]
Oh and updated to use snake-case as suggested

**Comment by mark-lunarg** [comment]
@souravpNV, the other PR (#1725) replaces this one entirely, yes?

**Comment by souravpNV** [comment]
Well, that was not my intention, #1725 was just meant to have test changes while this one to cater for validation layer changes.
This one is reflected properly, in #1725, only change that needs to be merged is test related change, the rest of that should be ignored.

**Comment by mark-lunarg** [comment]
Could you create a new PR (or force-push to this one) with your layers commit and then your tests commit on top of Master?  Git has a steep learning curve, but this will come up again and again.  Also, look at the failures in the other PR, at least some of those may need to be addressed in the new branch.

**Comment by souravpNV** [comment]
Thanks @mark-lunarg  I have created new PR #1731 and #1730.

**Comment by mark-lunarg** [comment]
Replaced by #1730 and #1731.

**Pull request closed without merging**